spec/integration/api_spec.rb

```diff
@@ -44,8 +44,22 @@
       @class.should respond_to(:search_for)
     end
 
-    it "should return a ActiveRecord::Relation instance" do
+    it "should return a ActiveRecord::Relation instance with no arguments" do
+      @class.search_for.should be_a(ActiveRecord::Relation)
+    end
+
+    it "should return a ActiveRecord::Relation instance with one argument" do
       @class.search_for('query').should be_a(ActiveRecord::Relation)
     end
+
+    it "should return a ActiveRecord::Relation instance with two arguments" do
+      @class.search_for('query', {}).should be_a(ActiveRecord::Relation)
+    end
+
+    it "should respect existing scope" do
+      @class.create! field: 'a'
+      record = @class.create! field: 'ab'
+      @class.where(field: 'ab').search_for('field ~ a').should eq([record])
+    end
```

Comment:
Thanks, I see your point. I've moved it into a new scope_spec.rb file and updated the PR.